### PR TITLE
arch/x86/ia32: Reserve GDT entries for ISH AON

### PR DESCRIPTION
**Pull request by leifuzhao**
The ISH power management code needs to reserve three GDT entry place holders which are filled during runtime. These three GDT entries are necessary for ISH AON to work together with ISH main firmware.

### CODE REVIEW

File: include/zephyr/arch/x86/ia32/scripts/shared_kernel_pages.ld [hunk 1]
```diff
@@ -48,6 +48,9 @@
 #endif /* CONFIG_X86_USERSPACE */
 	. += (GDT_NUM_ENTRIES + GDT_NUM_TLS_ENTRIES) * 8;
 #endif /* LINKER_ZEPHYR_FINAL */
+#if defined(CONFIG_SOC_FAMILY_INTEL_ISH) && defined(CONFIG_PM)
```
Comment (jhedberg) [inline include/zephyr/arch/x86/ia32/scripts/shared_kernel_pages.ld:51]: There's no `CONFIG_SOC_FAMILY_INTEL_ISH` in the upstream tree. This patch should probably be combined with a PR that adds the SoC.

Comment (leifuzhao) [reply include/zephyr/arch/x86/ia32/scripts/shared_kernel_pages.ld]: Done. Changed to use generic config way to set the reserved GDT entries, please review.

File: include/zephyr/arch/x86/ia32/scripts/shared_kernel_pages.ld [hunk 2]
```diff
@@ -48,6 +48,7 @@
 #endif /* CONFIG_X86_USERSPACE */
 	. += (GDT_NUM_ENTRIES + GDT_NUM_TLS_ENTRIES) * 8;
 #endif /* LINKER_ZEPHYR_FINAL */
+	. += CONFIG_GDT_RESERVED_NUM_ENTRIES * 8;
```
Comment (MaureenHelm) [inline include/zephyr/arch/x86/ia32/scripts/shared_kernel_pages.ld:51]: Should this be inside the `#ifdef LINKER_ZEPHYR_FINAL..#endif` block like the other gdt entries?

Comment (leifuzhao) [reply include/zephyr/arch/x86/ia32/scripts/shared_kernel_pages.ld:50]: Done. Put is into LINER_ZEPHYR_FINAL block same as other gdt entries, please review.

**Review by nashif — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**